hive/lib/src/sentry_hive_impl.dart

```diff
@@ -0,0 +1,203 @@
+import 'dart:typed_data';
+
+import 'package:hive/hive.dart';
+import 'package:meta/meta.dart';
+import 'package:sentry/sentry.dart';
+import 'sentry_box.dart';
+import 'sentry_lazy_box.dart';
+import 'sentry_hive_interface.dart';
+
+import 'default_compaction_strategy.dart';
+import 'default_key_comparator.dart';
+import 'sentry_span_helper.dart';
+
+/// @nodoc
+@internal
+class SentryHiveImpl implements SentryHiveInterface {
+  @internal
+  // ignore: public_member_api_docs
+  static const dbOp = 'db';
+
+  @internal
+  // ignore: public_member_api_docs
+  static const dbSystemKey = 'db.system';
+  @internal
+  // ignore: public_member_api_docs
+  static const dbSystem = 'noSQL';
```

Comment:
One last thing:

this should match ideally one of [OpenTelemetry specification](https://github.com/electrum/opentelemetry-specification/blob/6c5099addd0162fd74be1a2782bcb8cd8be0161a/specification/trace/semantic_conventions/database.md). Not sure which would fit the best though, maybe `other_sql` ? there's no info about key value storage dbs

Reply:
Changed to `nosql` Maybe @stefanosiano has an opinion here?

Reply:
```This custom value MUST be the name of the DBMS in lowercase```

we could use `hive` but it is already reserved for Apache Hive. I'd suggest something like `flutterhive` or `flutter_hive`